sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1506,6 +1545,52 @@ def process_bundle(self, inputs, expected_outputs):
     return result, split_results
 
 
+class ParallelBundleManager(BundleManager):
+
+  def _check_inputs_split(self, expected_outputs):
+    # We skip splitting inputs when timer is set, because operations are not
+    # triggered until we sent inputs for timers.
+    for _, pcoll_id in expected_outputs.items():
+      kind = split_buffer_id(pcoll_id)[0]
+      if kind in ['timers']:
+        return False
+
+    return True
+
+  def process_bundle(self, inputs, expected_outputs, num_workers=None):
+    num_workers = num_workers or self._num_workers
+    param_list = []
+
+    if self._check_inputs_split(expected_outputs):
+      for name, input in inputs.items():
+        for part in input.partition(num_workers):
+          param_list.append(({name : part}, expected_outputs))
```

Comment:
OK, I think this is where the bug above is. Timers are a case where there is more than one set of inputs, but this here adds a new element to param list for every input for every part. 

Instead, I think you need something like

```
partitioned_inputs = [{} for _ in range(num_workers)]
for name, input in inputs.items():
  for ix, part in input.partition(num_workers):
    partitioned_inputs[ix][name] = part
```

Reply:
This worked fantastically!!